agent/agent.go

```diff
@@ -699,25 +709,26 @@ func (a *Agent) startTask(ctx context.Context, tc *taskContext) (status string)
 
 	// notify API server that the task has been started.
 	tc.logger.Execution().Info("Reporting task started.")
-	if err = a.comm.StartTask(innerCtx, tc.task); err != nil {
+
+	if err := a.comm.StartTask(innerCtx, tc.task); err != nil {
```

Comment:
Since we aren't absorbing some things that apply only to pre and main (like the stats collector) in `runPreAndMain`, can we make a small helper function to also wrap the setup logic in `runPreAndMain`?

Reply:
Is there enough that we can move over that would make it worthwhile? It seems like it would just be the stats collector.

Reply:
Hmm, I guess it's not that much setup logic now that stuff has moved out. Never mind!